### ISSUE TITLE
pipeline @param.output usage

### ISSUE THREAD

**Issue by kcpevey** [pyviz/param#349]
This is a little strangeness. I'm not sure if this is anticipated behavior. If so, then it should be documented.  
  
this works (passes the anticipated owi_obj):  
```
@param.output(owi_obj=winds.OWI)
def output(self):  
        return self.execute()
```
This doesn't work (passes owi_obj as None):  
```
@param.output(owi_obj=winds.OWI)
def owi_obj(self):  
        return self.execute()
```
And this doesn't work (passes owi_obj as None):  
```
@param.output()
def owi_obj(self):  
        return self.execute()
```

**Comment by jbednar** [comment]
> This doesn't work (passes owi_obj as None):
```
@param.output(owi_obj=winds.OWI)
def owi_obj(self):  
        return self.execute()
```
Shouldn't this be the same as the following, and if so, does that work?
```
@param.output(winds.OWI)
def owi_obj(self):  
        return self.execute()
```

**Comment by ppwadhwa** [comment]
@kcpevey Can you summarize what changes we should include in the docs?  Thank you!
